### PR TITLE
ITransformable sample

### PR DESCRIPTION
**Pull request by mjkkirschner**
This PR adds a sample utilizing both `periodic` update and`ITransformable`

The sample builds a class that implements `IGraphicItem` but uses reflection to call `SetTransform` on the render package - an api added in 1.1 Dynamo.

A simple run periodic node is used to alter the displayed geometry without recomputing the geometry.

a sample file using the nodes is also added.

Reviewers:
@ikeough (if you have the bandwidth)
@saintentropy

### CODE REVIEW

File: src/SampleLibraryZeroTouch/Examples/TransformableExample.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        /// <param name="transform"></param>
+        /// <returns></returns>
+        public TransformableExample TransformObject(CoordinateSystem transform)
+        {
+            this.transform = transform;
+            return this;
+        }
+
+        /// <summary>
+        /// this method is actually called by Dynamo when it attempts to render the TransformableExample
```
Comment (ikeough) [inline src/SampleLibraryZeroTouch/Examples/TransformableExample.cs]: "This method..."

File: src/SampleLibraryZeroTouch/Examples/TransformableExample.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        [Autodesk.DesignScript.Runtime.IsVisibleInDynamoLibrary(false)]
+        public TransformableExample(Geometry geo)
+        {
+            Geometry = geo;
+            //initial transform is just at the origin
+            transform = CoordinateSystem.ByOrigin(0, 0, 0);
+        }
+
+     /// <summary>
+     /// An object which knows how to draw itself in the background preview and uses a transform to take
```
Comment (ikeough) [inline src/SampleLibraryZeroTouch/Examples/TransformableExample.cs]: This not should go on the class. 

Is transformability part of `IGraphicItem`?

File: src/SampleLibraryZeroTouch/Examples/TransformableExample.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        }
+
+     /// <summary>
+     /// An object which knows how to draw itself in the background preview and uses a transform to take 
+     /// advantage of the GPU to alter that background visualization. The original geometry remains unaltered,
+     /// only the visualization is transformed.
+     /// </summary>
+     /// <param name="geometry"> a geometry object</param>
+     /// <returns></returns>
+        public static TransformableExample byGeometry(Autodesk.DesignScript.Geometry.Geometry geometry)
```
Comment (ikeough) [inline src/SampleLibraryZeroTouch/Examples/TransformableExample.cs]: `ByGeometry(...)`

File: src/SampleLibraryZeroTouch/Examples/TransformableExample.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            var newTransformableThing = new TransformableExample(geometry);
+            return newTransformableThing;
+        }
+
+        /// <summary>
+        /// this method sets the transform on the object and returns a reference to the object so
```
Comment (ikeough) [inline src/SampleLibraryZeroTouch/Examples/TransformableExample.cs]: "This method..."

File: src/SampleLibraryZeroTouch/Examples/TransformableExample.cs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        /// the tessellate method is called and the new visualization shows in the background preview
```
Comment (ikeough) [inline src/SampleLibraryZeroTouch/Examples/TransformableExample.cs]: "...shows in the background preview."

File: src/SampleLibraryZeroTouch/Examples/TransformableExample.cs (hunk 2 above)
Comment (mjkkirschner) [reply src/SampleLibraryZeroTouch/Examples/TransformableExample.cs]: no, it's part of `ITransformable` which is applied to `RenderPackages` - I will turn this summary into a simple comment.

**Comment by mjkkirschner** [comment]
thanks ian, will make changes

File: src/SampleLibraryZeroTouch/Examples/TransformableExample.cs (hunk 2 above)
Comment (mjkkirschner) [reply src/SampleLibraryZeroTouch/Examples/TransformableExample.cs]: ~~wait, what shouldn't go on the class?~~

I think I see you were saying the summary should go on the class, but if thats the case it won't be exposed in Dynamo anywhere afaik.

**Comment by saintentropy** [comment]
LGTM

**Pull request merged**